crates/conjure-cp-core/src/solver/adaptors/rustsat/convs.rs

```diff
@@ -60,52 +60,55 @@ pub fn handle_atom(
                 todo!("Not Sure if we are handling Lits as-is or not..")
             }
             conjure_cp_core::ast::Atom::Reference(decl) => match &*(decl.name()) {
-                conjure_cp_core::ast::Name::User(n) => {
+                conjure_cp_core::ast::Name::User(_)
+                | conjure_cp_core::ast::Name::Represented(_)
+                | conjure_cp_core::ast::Name::Machine(_) => {
                     // TODO: Temp Clone
                     // let m = n.clone();
-                    let lit_temp: Lit = fetch_lit(n.to_string(), vars_added, inst);
+                    let lit_temp: Lit = fetch_lit(decl.name().clone(), vars_added, inst);
                     if polarity { lit_temp } else { !lit_temp }
                 }
-                _ => {
-                    todo!("Change Here for other types of vars")
-                }
+                _ => todo!("Not implemented yet"),
             },
         },
         _ => panic!("atomic expected"),
     }
 }
 
-pub fn fetch_lit(
-    symbol: String,
-    vars_added: &mut HashMap<String, Lit>,
-    inst: &mut SatInstance,
-) -> Lit {
-    if !vars_added.contains_key(&symbol) {
-        vars_added.insert(symbol.to_string(), inst.new_lit());
+pub fn fetch_lit(name: Name, vars_added: &mut HashMap<Name, Lit>, inst: &mut SatInstance) -> Lit {
+    if !vars_added.contains_key(&name) {
+        vars_added.insert(name.clone(), inst.new_lit());
     }
-    *(vars_added.get(&symbol).unwrap())
+    *(vars_added.get(&name).unwrap())
 }
 
 pub fn handle_disjn(
     disjn: &Expression,
-    vars_added: &mut HashMap<String, Lit>,
+    vars_added: &mut HashMap<Name, Lit>,
     inst_in_use: &mut SatInstance,
 ) {
-    let cl: Vec<Expression> = match disjn {
-        Expression::Or(_, vec) => Moo::unwrap_or_clone(Moo::clone(vec)).unwrap_list().unwrap(),
-        _ => panic!("Expected an 'Or' expression!"),
-    };
-
     let mut lits = Clause::new();
-    for literal in cl {
-        let lit: Lit = handle_lit(&literal, vars_added, inst_in_use);
-        lits.add(lit);
-    }
+
+    match disjn {
+        // treat ors and clauses the same
+        Expression::Or(_, vec) | Expression::Clause(_, vec) => {
```

Comment:
Ideally an `Or` would only be recieved by this if there is a bug in `TreeMorph`.
```suggestion

    match disjn {
        // treat ors and clauses the same
        Expression::Clause(_, vec) => {
            let cl = Moo::unwrap_or_clone(vec.clone()).unwrap_list().unwrap();
            for literal in cl {
                let lit: Lit = handle_lit(&literal, vars_added, inst_in_use);
                lits.add(lit);
            }
        }

        literal => {
            let lit: Lit = handle_lit(literal, vars_added, inst_in_use);
            lits.add(lit);
        }

        _ => bug!("Only expecting Expression::Clause here.")
    };
```

But again, we're not sure if Clause Expression is needed.

~ Shikhar-Srivastava-16
~ YehorBoiar